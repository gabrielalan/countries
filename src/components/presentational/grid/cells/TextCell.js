/**
 * This is the most basic type of cell used on grids
 * It turns everything into strings so no error will be thrown
 * when the children is some kind of object not accepted by React
 */
export default function TextCell({ children }) {
  return children && children.toString();
}