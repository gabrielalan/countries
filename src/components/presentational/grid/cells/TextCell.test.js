import React from 'react';
import renderer from 'react-test-renderer';
import TextCell from './TextCell';

describe('TextCell', () => {
  it('should render text normally', () => {
    const tree = renderer.create(<TextCell>text</TextCell>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render numbers normally', () => {
    const tree = renderer.create(<TextCell>{2.4}</TextCell>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should represent objects without errors', () => {
    const tree = renderer.create(<TextCell>{{ text: 'text' }}</TextCell>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should represent lists without errors', () => {
    const tree = renderer.create(<TextCell>{[1, 'text', {}]}</TextCell>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should throw errors when children is null', () => {
    const tree = renderer.create(<TextCell>{null}</TextCell>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
