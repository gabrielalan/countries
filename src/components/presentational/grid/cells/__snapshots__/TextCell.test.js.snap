// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextCell should render numbers normally 1`] = `"2.4"`;

exports[`TextCell should render text normally 1`] = `"text"`;

exports[`TextCell should represent lists without errors 1`] = `"1,text,[object Object]"`;

exports[`TextCell should represent objects without errors 1`] = `"[object Object]"`;

exports[`TextCell should throw errors when children is null 1`] = `null`;
